### PR TITLE
BRIDGE-3268: consolidate account deletion methods

### PR DESCRIPTION
**Pull request by alxdarksage**
There is now one canonical delete method in AccountService that handles clearing all the caches, the non-transactional DynamoDB tables, etc.

### CODE REVIEW

File: src/main/java/org/sagebionetworks/bridge/services/AccountService.java [hunk 1]
```diff
@@ -94,7 +84,7 @@ protected String generateGUID() {
      * Search for all accounts across apps that have the same Synapse user ID in common, 
      * and return a list of the app IDs where these accounts are found.
      */
-    public List<String> getAppIdsForUser(String synapseUserId) {
+    public List<String> getAppIdsForUser2(String synapseUserId) {
```
Comment (jnelsonjava) [inline src/main/java/org/sagebionetworks/bridge/services/AccountService.java]: Just curious what this rename is for? Are you differentiating from the same name method in AdminAccountService?

Comment (jnelsonjava) [reply src/main/java/org/sagebionetworks/bridge/services/AccountService.java]: Actually should this method just be deleted?

**Review by jnelsonjava — APPROVED** [review]
Approved but check out the commented method in AccountService. I'm pretty sure it can just be removed now.

File: src/main/java/org/sagebionetworks/bridge/services/AccountService.java (hunk 1 above)
Comment (alxdarksage) [reply src/main/java/org/sagebionetworks/bridge/services/AccountService.java]: Yes this was a step to ensuring all code was pointed at the same method in AdminAccountService; I will delete it.

**Pull request merged**